webdriver-spec.html

```diff
@@ -3220,35 +3222,26 @@ <h2 id="element-location-strategies">Element Location Strategies</h2>
 
   <p>Otherwise, return <var>element result</var>.
 
- <li><p>If <a>element</a> <a>is stale</a>,
+ <li><p>If <var>element</var> <a>is stale</a>,
   return <a>error</a> with <a>error code</a> <a>stale element reference</a>.
 
- <li><p>Let <var>name</var> be equal to "name" from <a>match a request</a>.
-   if <var>name</var> is undefined return <a>error</a> with <a>error code</a>
-   <a>invalid argument</a>.
-  <li>Let <var>result</var> be the result of the first appropriate step below:
-
-   <dl class=switch>
-     <dt><p>If <var>name</var> is a
-       <a href='https://html.spec.whatwg.org/#boolean-attribute'>boolean attribute</a>:
-     <dd><p>The first matching statement if the element
-       <a href=https://dom.spec.whatwg.org/#dom-element-hasattribute>has the attribute</a>:
-       <dl class=switch>
-         <dt><code>true</code>
-         <dd><p>'true'.
-         <dt>Otherwise
-         <dd><p><code>Null</code>.
-      </dl>
-      <dt><p>Otherwise:
-      <dd><p>The result of <a href=https://dom.spec.whatwg.org/#concept-element-attributes-get-by-name>
-          getting the attribute by name</a>.
-   </dl>
-   <li><p>Let <var>body</var> be a JSON Object
-    with the "<code>value</code>" member set to <var>result</var>.
-
-   <li><p>Return <a>success</a> with data <var>body</var>.
+ <li><p>If <var>name</var> is undefined,
+  return <a>error</a> with <a>error code</a> <a>invalid argument</a>.
+
+ <li><p>Let <var>attribute</var> be the result of
+  <a href=https://dom.spec.whatwg.org/#concept-element-attributes-get-by-name>getting an attribute</a>
+  given <var>name</var> and <var>element</var>.
+
+ <li><p>Let <var>result</var> be the value of <var>attribute</var>
```

Comment:
I’m not really sure what you’re proposing here.  That **Get Element Attribute** duplicate the behaviour of **Get Element Property** _specifically_ when interacting with form control elements?  All elements?

I think my primary counterargument is that attributes is a DOM concept, and the DOM doesn’t say anything about disabledness/checkedness of HTML elements.  I don’t think it’s a particularly good idea if we rocked this idea, despite that HTML reflects changes to the property to the DOM.

Since it’s HTML that [defines the **disabled** concept](https://html.spec.whatwg.org/#concept-fe-disabled), what users want also isn’t `hasAttribute("disabeld")`, although in the particular case of calling that on form control elements it would _effectively_ give you the same result.

I think making the change I think you’re suggesting would lead to a situation where WebDriver has a special definition of what it means to be a content attribute, different to that of DOM and HTML.  It would effectively mean that if users actually wanted the actual, _correct_ value of the content attribute, they’d have to call **Execute Script**, asking for `arguments[0].getAttribute("disabled")`, which yet again proves why we shouldn’t overload the “attribute” concept.  It’s hard enough to explain what **Get Element Attribute** does in Selenium WebDriver as it is.

I’d rather we keep the primitives close to existing web platform technologies instead of inventing our own behaviour and _especially_ overload existing terminology.  If local ends wish to implement a behaviour that diverges from the web platform, they are able to do so with the primitives we provide them with.